sdk/identity/azure-identity/azure/identity/_credentials/linux_vscode_adapter.py

```diff
@@ -58,22 +66,17 @@ def _get_refresh_token(service_name, account_name):
     if not _libsecret:
         return None
 
-    # _libsecret.secret_password_lookup_sync raises segment fault on Python 2.7
-    # temporarily disable it on 2.7
-    import sys
-
-    if sys.version_info[0] < 3:
-        raise NotImplementedError("Not supported on Python 2.7")
-
-    if sys.version_info >= (3, 8):
-        raise NotImplementedError("Not supported")
-
     err = ct.c_int()
-    schema = _libsecret.secret_schema_new(
-        _c_str("org.freedesktop.Secret.Generic"), 2, _c_str("service"), 0, _c_str("account"), 0, None
-    )
+    attributes = [_SECRET_SCHEMA_ATTRIBUTE(_c_str("service"), 0), _SECRET_SCHEMA_ATTRIBUTE(_c_str("account"), 0)]
+    pattributes = (_SECRET_SCHEMA_ATTRIBUTE * 2)(*attributes)
+    schema = _SECRET_SCHEMA()
+    pschema = _PSECRET_SCHEMA(schema)
+    ct.memset(pschema, 0, ct.sizeof(schema))
+    setattr(schema, "name", _c_str("org.freedesktop.Secret.Generic"))
```

Comment:
Is `setattr` necessary here?

Reply:
This is from what I know, how to set value for c-like structures.

Reply:
The difference is ct.memset(pschema, 0, ct.sizeof(schema))
 by calling
schema = _SECRET_SCHEMA(name=_c_str("org.freedesktop.Secret.Generic"), flags=2, attributes=pattributes) only,
it is not guaranteed the rest of the memory is set to 0

Reply:
I don't see how that requires `setattr`. Wouldn't this still work?
```py
schema = _SECRET_SCHEMA()
pschema = _PSECRET_SCHEMA(schema)
ct.memset(pschema, 0, ct.sizeof(schema))
schema.name = _c_str("org.freedesktop.Secret.Generic")
```
Also, why zero the schema's memory? You assign all its fields.

Reply:
attributes is an array which can be more than 2. In our case, we make 2 in the definition but no harm to set it. We only call it once so the cost is negligible